src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs

```diff
@@ -56,198 +30,27 @@ public static IHealthChecksBuilder AddMongoDb(
     /// <returns>The specified <paramref name="builder"/>.</returns>
     public static IHealthChecksBuilder AddMongoDb(
         this IHealthChecksBuilder builder,
-        string mongodbConnectionString,
-        string mongoDatabaseName,
+        Func<IServiceProvider, IMongoClient>? clientFactory = default,
+        Func<IServiceProvider, string>? databaseNameFactory = default,

```

Comment:
Should we also have an overload that instead returns an `IMongoDatabase` instead of both of these?

Reply:
Good point. I've added such an overload, but I could not have made the factory optional argument, as compiler would not be able to choose what `.AddMongo()` means. I expect `IMongoDatabase` to be less frequently used, so I've made the `Func<IServiceProvider, IMongoDatabase> dbFactory` argument mandatory.

Reply:
> so I've made the Func<IServiceProvider, IMongoDatabase> dbFactory argument mandatory.

Yep. that sounds right to me.